### DISCUSSION TITLE
Any guide on how to proper integrate the latest Elsa Workflows on Abp commercial? or AspNet Zero?

### DISCUSSION THREAD

**Discussion by robson-paproski** [elsa-workflows/elsa-core#5214] [Q&A]
Need some help, There is any guide available to integrate the newest Elsa Workflows with Asp Net Zero, or since its blazor, directly with Abp framework/Abp Commercial?

**Comment by glime-ai** [comment]
As of my knowledge cutoff date in early 2023, there is no specific guide available for integrating Elsa Workflows with Asp Net Zero or directly with the ABP framework/ABP Commercial. However, since Elsa Workflows is a .NET-based workflow engine, it can be integrated into any .NET application, including those built with Asp Net Zero or the ABP framework.

To integrate Elsa Workflows into your Asp Net Zero or ABP framework application, you would typically follow the general steps for integrating Elsa into an ASP.NET Core application. This would involve:

1. Adding the necessary Elsa NuGet packages to your project.
2. Configuring services in the `Startup.cs` or `Program.cs` file (depending on the version of .NET you are using).
3. Optionally setting up Elsa Dashboard if you need a UI to manage workflows.
4. Creating and registering your workflows within the application.

Since Asp Net Zero and the ABP framework have their own set of conventions and structures, you may need to adapt the integration steps to fit within those frameworks. It would be beneficial to have a good understanding of how dependency injection, middleware, and modules work within Asp Net Zero or ABP to ensure a smooth integration.

If you require a more specific guide, you might consider reaching out to the Elsa Workflows community or the Asp Net Zero/ABP community for assistance. Additionally, you could look into the possibility of creating a custom module or service within your application that encapsulates the Elsa Workflows integration.

For more detailed instructions on integrating Elsa Workflows into an ASP.NET Core application, you can refer to the official Elsa documentation and adapt the steps as necessary for your specific framework.

**Comment by jacekmichalski** [comment]
@robson-paproski did you manage to integrate elsa with abp framework? If yes, can you give us the steps how to or any hints? Thanks in advance

**Comment by sfmskywalker** [comment]
There is currently no maintained Elsa guide specifically for ABP Commercial or ASP.NET Zero. The supported starting point is the generic Elsa Server / ASP.NET Core setup, which you can adapt into an ABP module.

For ABP/OpenIddict authentication, see the current guidance in [#6773](https://github.com/elsa-workflows/elsa-core/discussions/6773): let the host own authentication, avoid Elsa's default JWT/API-key setup when it conflicts, and register the Elsa endpoint policies required by the host.

The generic server guide and sample repository are the best starting points. If you can share your ABP version, .NET version, hosting model, authentication setup, and the exact integration error, we can help narrow it down. [#7122](https://github.com/elsa-workflows/elsa-core/issues/7122) tracks a similar ASP.NET Zero integration problem.
